### DISCUSSION TITLE
RT: eassynth backend should not include all the sonivox library sources

### DISCUSSION THREAD

**Discussion by pedrolcl** [pedrolcl/drumstick#8] [Announcements]
This comes from [SourceForge Ticket 44](https://sourceforge.net/p/drumstick/bugs/44/).

Currently, the eassynth RT backend includes the sonivox library sources, which are compiled and statically linked. Instead, the build system should search for an external [sonivox library](https://github.com/pedrolcl/sonivox) like any other dependency, and dynamically link the shared library when found. The RT backends are currently used by these applications:

* [drumstick-vpiano](https://github.com/pedrolcl/drumstick/tree/master/utils/vpiano) (from the drumstick utils)
* [VMPK](https://github.com/pedrolcl/VMPK)
* [dmidiplayer](https://github.com/pedrolcl/dmidiplayer)

The [sonivox library](https://github.com/pedrolcl/sonivox) ~~is not yet released, but I'm working on it. It is only a few days away.~~ is already released. Please find [here](https://github.com/pedrolcl/sonivox/releases/tag/v3.6.10.14) the source tarballs.

There are two other projects/prototypes that use this library in my portfolio. They embed it as a git submodule, or find it as an external dependency:

* [Linux-SonivoxEAS](https://github.com/pedrolcl/Linux-SonivoxEas)
* [multiplatform-sonivoxeas](https://github.com/pedrolcl/multiplatform-sonivoxeas)

I prefer to have all the code in a single place, avoiding duplication. But it is not only my personal convenience, Linux users would have a single copy of the library installed in their systems, which will be easier to patch or replace by the distribution vendors when needed. Of course this would be not possible if their only choice is to install the applications as an AppImage, or a  Flatpak.

This is currently pending on implementation.

**Comment by dvzrv** [comment]
> The [sonivox library](https://github.com/pedrolcl/sonivox) is not yet released, but I'm working on it. It is only a few days away.

As long as it is released and buildable before drumstick's next release, I have no issue with adding that as a new dependency!

**Reply by pedrolcl** [reply]
David, are you already maintaining 885 Arch packages? really? Maybe I'm wrong, but if not then I understand that one more doesn't really matter.

**Comment by pedrolcl** [comment]
There is another [discussion thread](https://github.com/pedrolcl/sonivox/discussions/1) about the release of the sonivox library.

**Comment by pedrolcl** [comment]
Drumstick v2.7.0 has been [released](https://sourceforge.net/projects/drumstick/files/2.7.0/).
